Sources/ConfidenceProvider/Apply/FlagApplierWithRetries.swift

```diff
@@ -37,68 +37,66 @@ final class FlagApplierWithRetries: FlagApplier {
 
     public func apply(flagName: String, resolveToken: String) async {
         let applyTime = Date.backport.now
-        let (data, added) = await cacheDataInteractor.add(
+        async let (data, added) = await cacheDataInteractor.add(
```

Comment:
isn't here the problem that the add function in the cacheDataInteractor is not defined async? then if it's defined as async we can just await it normally here?

Reply:
can we try making the cache interactor add funciton async and using await when accessing cache in [here](https://github.com/spotify/confidence-openfeature-provider-swift/blob/9d8d1732f8224b719afd0ee20b9cbcf6b3b6e8a2/Sources/ConfidenceProvider/Cache/CacheDataInteractor.swift#L12) ?